File: Data Structures & Algorithms/binary-search/submission-1.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
    int l , r , m;
    l = 0; r = nums.size() - 1; 
    m = l + (r - l) / 2;
    while(l <= r){
         if(nums[m] > target){
             r = m - 1;
             m = l + (r - l) / 2;
         }
         else if(nums[m] < target){
             l = m + 1;
             m = l + (r - l) / 2;
         }
         else{
             return m;
         }
    } 
    return -1;
 }
};

File: Data Structures & Algorithms/move-zeroes/submission-1.java
class Solution {
    public void moveZeroes(int[] nums) {
        int l = 0 , r = 0;
        while(r < nums.length && l < nums.length){
            if(nums[l] != 0){
                while(l < nums.length&&nums[l] != 0){
                    l+=1;
                }
            }
            if(nums[r] != 0 && r > l){
                nums[l] = nums[r];
                nums[r] = 0;
            }
            r+=1;
        } 
    }
}